packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -78,14 +130,28 @@ export class SharedTreeCore<TChangeRebaser extends ChangeRebaser<any, any, any>>
         const newLocalRevisionHead = fail("TODO: rebase local changes onto new sequencedRevision");
         this.updateLocalState(newLocalRevisionHead);
     }
+
     protected onDisconnect() {
         throw new Error("Method not implemented.");
     }
+
     protected applyStashedOp(content: any): unknown {
         throw new Error("Method not implemented.");
     }
 
-    // TODO: custom getGCData.
+    public getGCData(fullGC?: boolean): IGarbageCollectionData {
+        const gcNodes: IGarbageCollectionData["gcNodes"] = {};
+        for (const summaryElement of this.summaryElements) {
+            for (const [id, routes] of Object.entries(summaryElement.getGCData(fullGC).gcNodes)) {
+                gcNodes[id] ??= [];
+                gcNodes[id].push(...routes);
```

Comment:
if the number of routes exceeds the limit for the number of arguments, this will fail. I suspect these lists could get long, so might want to use a loop here instead.

Reply:
I didn't realize that there was a limit to `push(...)`. Or is there a limit to the spread operator in general? Is that what that whiteboard bug was about? I'll change it.